pkg/cache/v3/simple.go

```diff
@@ -606,6 +609,20 @@ func (cache *snapshotCache) Fetch(ctx context.Context, request *Request) (Respon
 	return nil, fmt.Errorf("missing snapshot for %q", nodeID)
 }
 
+// GetNodeIDs gets the all node ID in cache.
+func (cache *snapshotCache) GetNodeIDs() []string {
+	cache.mu.RLock()
+	defer cache.mu.RUnlock()
+
+	nodeIDs := []string{}
```

Comment:
```suggestion
	nodeIDs := make([]string, 0, Len(cache.snapshots))
```

Micro-optimize the allocation :)

Reply:
In fact, you could copy the `GetStatusKeys` implementation directly here :)